### ISSUE TITLE
Can trust an out of date certificate 

### ISSUE THREAD

**Issue by turtle0x1** [lxc/lxd#10041]
If you generate a certificate with an expiry date in the past you can push it to LXD, API requests using that cert are rejected but might considered a bug that it can be pushed in the first place.

**Comment by stgraber** [comment]
Yeah, probably reasonable to have LXD check the expiry prior to adding to the trust store. If anything it may make people realize that they have a system that's in dire need of a working NTP setup :)

**Comment by turtle0x1** [comment]
I know I pointed this out but;

As this has no security implications, I wonder if all this really does is ruin the dev experience for no real practical gain.

I discovered this because I needed to test the out of date certificate situation, from a user complaint.

The fact I could push it using the typical route was very helpful, hacking it together would be hard work. The alternative, deploying a certificate that expires in 1 day and waiting 24 hours would have annoyed no end (as would changing the system time).

**Comment by turtle0x1** [comment]
Could it maybe be a `warning`? `Cert pushed that is out of date`?

**Comment by stgraber** [comment]
I think an error is appropriate in this case. It certainly makes it harder to test the certificate out of date case by having to either use `lxd sql global` to update the DB record (and reload LXD) or play with the system time to let it load, but that seems reasonable to me overall.
